network/src/lib.rs

```diff
@@ -828,37 +825,26 @@ impl PolkadotProtocol {
 	/// This should be called by a collator intending to get the locally-collated
 	/// block into the hands of validators.
 	/// It also places the outgoing message and block data in the local availability store.
-	pub fn add_local_collation(
+	pub async fn add_local_collation(
 		&mut self,
 		ctx: &mut dyn Context<Block>,
 		relay_parent: Hash,
 		targets: HashSet<ValidatorId>,
 		collation: Collation,
 		outgoing_targeted: OutgoingMessages,
-	) -> impl futures::future::Future<Item = (), Error=()> {
+	) {
 		debug!(target: "p_net", "Importing local collation on relay parent {:?} and parachain {:?}",
 			relay_parent, collation.info.parachain_index);
 
-		let res = match self.availability_store {
-			Some(ref availability_store) => {
-				let availability_store_cloned = availability_store.clone();
-				let collation_cloned = collation.clone();
-				Either::A((async move {
-						let _ = availability_store_cloned.make_available(av_store::Data {
-							relay_parent,
-							parachain_id: collation_cloned.info.parachain_index,
-							block_data: collation_cloned.pov.block_data.clone(),
-							outgoing_queues: Some(outgoing_targeted.clone().into()),
-						}).await;
-					}
-				)
-					.unit_error()
-					.boxed()
-					.compat()
-					.then(|_| Ok(()))
-				)
-			}
-			None => Either::B(futures::future::ok::<(), ()>(())),
+		if let Some(ref availability_store) = self.availability_store {
+			let availability_store_cloned = availability_store.clone();
```

Comment:
If I am not mistaken, I believe that this `.clone()` can now be removed, I've added it to be consumed by the `async move`.